kratos/utilities/entities_utilities.cpp

```diff
@@ -12,16 +12,199 @@
+std::string EntitityIdentifier<TEntity>::GetEntityTypeName() const
+{
+    std::string entity_type_name;
+    if constexpr (std::is_same<TEntity, Element>::value) {
+        entity_type_name = "Element";
+    } else if constexpr (std::is_same<TEntity, Condition>::value) {
+        entity_type_name = "Condition";
+    } else {
+        static_assert(std::is_same<TEntity, Element>::value || std::is_same<TEntity, Condition>::value, "Unsupported entity type. Only element and condition are supported.");
+    }
+    return entity_type_name;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+void EntitityIdentifier<TEntity>::GenerateSingleType(const std::string& rName)
 {
+    // Check the type of entity considered
+    const std::string entity_type_name = GetEntityTypeName();
+
+    // Some values need to be mandatorily prescribed since no meaningful default value exist. For this reason try accessing to them so that an error is thrown if they don't exist
+    KRATOS_ERROR_IF(rName != "" && !KratosComponents<TEntity>::Has(rName)) << entity_type_name << " name not found in KratosComponents<" << entity_type_name << "> -- name is " << rName << std::endl;
+    const auto& r_entity = KratosComponents<TEntity>::Get(rName);
+    mpPrototypeEntity = r_entity.Create(0, r_entity.pGetGeometry(), nullptr);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+void EntitityIdentifier<TEntity>::GenerateMultipleTypes(const std::string& rName)
+{
+    // Check the type of entity considered
+    const std::string entity_type_name = GetEntityTypeName();
+
+    // Split the name
+    const auto splitted_names = StringUtilities::SplitStringByDelimiter(rName, ';');
+    for (auto& r_entity_name : splitted_names) {
+        // Some values need to be mandatorily prescribed since no meaningful default value exist. For this reason try accessing to them so that an error is thrown if they don't exist
+        KRATOS_ERROR_IF(r_entity_name != "" && !KratosComponents<TEntity>::Has(r_entity_name)) << entity_type_name << " name not found in KratosComponents<" << entity_type_name << "> -- name is " << r_entity_name << std::endl;
+        const auto& r_ref_entity = KratosComponents<TEntity>::Get(r_entity_name);
+        const auto& r_reference_geometry = r_ref_entity.GetGeometry();
+        const auto& r_reference_geometry_type = r_reference_geometry.GetGeometryType();
+        mTypes.insert({r_reference_geometry_type, &KratosComponents<TEntity>::Get(r_entity_name)});
+    }
+
+    // Set the definition type
+    mDefinitionType = EntitiesUtilities::DefinitionType::Multiple;
+
+    // Assign default prototype
+    const auto* p_entity = mTypes.begin()->second;
```

Comment:
this is wrong again... if i have more than one type i am creating it here with the wrong type.

why do you need mpPrototypeEntity if you have multiple types?

Reply:
but you are verifying every time if the type is correct...

Reply:
Maybe will be some corner cases, but I am quite confident that in most cases it will be like this. @roigcarlo opinion?

Reply:
Yes it is threadsafe if you copy (private) the EntityIdentifier, see https://github.com/KratosMultiphysics/Kratos/pull/10726, where is actually used